libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -66,18 +88,26 @@ bool IRGeneratorForStatements::visit(Assignment const& _assignment)
 
 	_assignment.rightHandSide().accept(*this);
 
-//	solUnimplementedAssert(
-//		*_assignment.rightHandSide().annotation().type == *_assignment.leftHandSide().annotation().type,
-//		"Type conversion not yet implemented"
-//	);
 	// TODO proper lvalue handling
 	auto const& identifier = dynamic_cast<Identifier const&>(_assignment.leftHandSide());
 	string varName = m_context.variableName(dynamic_cast<VariableDeclaration const&>(*identifier.annotation().referencedDeclaration));
-	m_code << varName << " := " << m_context.variable(_assignment.rightHandSide()) << "\n";
+
+	m_code << varName << " := ";
+
+	Type const& from = *_assignment.rightHandSide().annotation().type;
+	Type const& to   = *_assignment.leftHandSide().annotation().type;
+
+	appendConvertIfRequired(from, to, m_context.variable(_assignment.rightHandSide()));
```

Comment:
Perhaps we could pass just an expression to this function and it would automatically retrieve the variable name and the from type?

Reply:
I like this idea!